plugins/modules/user_role_workflow_manager.py

```diff
@@ -1043,8 +1043,8 @@ def valid_role_config_parameters(self, role_config):
         error_messages = []
 
         role_name = role_config.get("role_name")
-        role_name_regex = re.compile(r"^[A-Za-z0-9_-]+$")
-        role_name_regex_msg = "must only contain letters, numbers, underscores and hyphens and should not contain spaces or other special characters."
+        role_name_regex = re.compile(r"^[a-zA-Z0-9._-]{,25}$")
+        role_name_regex_msg = "Role names must be 0 to 25 characters long and should contain only letters, numbers, dots, underscore, and hyphen."
```

Comment:
0 to 25 is correct? can we say periods or full stops instead of dots?

role_name_regex_msg = "Role names must be 1 to 25 characters long and should contain only letters, numbers, periods, underscores, and hyphens."

Reply:
Updated the code accordingly.